src/uri.rs

```diff
@@ -179,6 +175,39 @@ impl Uri {
     }
 }
 
+fn parse_scheme(s: &str) -> Option<usize> {
+    s.find(':')
+}
+
+fn parse_authority(s: &str) -> Option<usize> {
+    let v: Vec<&str> = s.split("://").collect();
+    let auth = v.last().unwrap()
```

Comment:
Well, since this is meant to be an option, it may be a good idea to use the `?` operator there to return the `None` value instead of panicking.